src/frontend/src/components/authenticateBox.ts

```diff
@@ -255,22 +255,25 @@ const page = (slot: TemplateResult) => {
 
 // Start the "remote device" add flow by prompting for an anchor, and then
 // adding a device to that anchor.
-const addRemoteDevice = async (connection: Connection, userNumber?: bigint) => {
+const addRemoteDevice = async (
+  connection: Connection,
+  userNumber_?: bigint
+) => {
   // Prompt the user for an anchor (only used if we don't know the anchor already)
   const askUserNumber = async () => {
-    const _userNumber = await promptUserNumber({
+    const userNumber = await promptUserNumber({
       title: "Add a Trusted Device",
       message: "What’s your Identity Anchor?",
     });
-    if (_userNumber === "canceled") {
+    if (userNumber === "canceled") {
       // TODO L2-309: do this without reload
       return window.location.reload() as never;
     }
 
-    return _userNumber;
+    return userNumber;
   };
 
-  const _userNumber = userNumber ?? (await askUserNumber());
+  const userNumber = userNumber_ ?? (await askUserNumber());
```

Comment:
technically if you inline the check I think you would maybe not even need `userNumber_` here right?

```
const addRemoteDevice = async (
  connection: Connection,
  userNumber?: bigint
) => {
  // Prompt the user for an anchor (only used if we don't know the anchor already)
  const askUserNumber = async () => {
    const userNumber = await promptUserNumber({
      title: "Add a Trusted Device",
      message: "What’s your Identity Anchor?",
    });
    if (userNumber === "canceled") {
      // TODO L2-309: do this without reload
      return window.location.reload() as never;
    }

    return userNumber;
  };

  return registerTentativeDevice(userNumber ?? (await askUserNumber()), connection);
};
```

(not a call to action or comment, meant as question only)

Reply:
yes that's correct! let me give this a think, might be less error prone

Reply:
@peterpeterparker I've changed it to do like you suggest, thanks! 👍